src/test/java/echoserver/EchoServerServiceTest.java

```diff
@@ -0,0 +1,88 @@
+package echoserver;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.ArrayList;
+
+
+
+public class EchoServerServiceTest {
+
+    private ArrayList<ClientConnection> clients = new ArrayList<ClientConnection>();
+    private EchoServerService testServer;
+    private ServerSocket hostSocket;
+
+    class FakeHostServer extends HostServer {
+        FakeHostServer(ServerSocket hostSocket) {
+            super(hostSocket);
+        }
+
+        public ClientConnection listenForClientConnection() throws IOException {
+            return clients.remove(0);
+        }
+    }
+
+    class FakeClientConnection extends ClientConnection{
+         FakeClientConnection(FakeSocket socket){
+            super(socket);
+        }
+    }
+
+    @Before
+    public void setupService() throws IOException {
+        hostSocket = new ServerSocket(5000);
+        FakeHostServer host = new FakeHostServer(hostSocket);
+
+        Logger echoLogger = new Logger(new Console(new ByteArrayOutputStream()));
+        testServer = new EchoServerService(host, echoLogger);
+    }
+
+    @Test
+    public void whenTheServerIsRunning_multipleClientsCanConnectAtOnce() throws IOException{
```

Comment:
This test feels brittle to me. It is testing the service, but it depends on many lower level details. For instance, does it need to use sockets for verification?